File: Backend/Model/Anime.cs
namespace Backend.Model
{
    public class Anime
    {
        public int Id { get; set; }
        public string? Title { get; set; }
        public virtual MainPicture? Main_picture { get; set; }
        public virtual AlternativeTitles? Alternative_titles { get; set; }
        public string? Start_date { get; set; }
        public string? End_date { get; set; }
        public string? Synopsis { get; set; }
        public float? Mean { get; set; }
        public int? Rank { get; set; }
        public int? Popularity { get; set; }
        public int? Num_list_users { get; set; }
        public int? Num_scoring_users { get; set; }
        public string? Nsfw { get; set; }
        public virtual ICollection<Genre>? Genres { get; set; }
        public DateTime? Created_at { get; set; }
        public DateTime? Updated_at { get; set; }
        public string? Media_type { get; set; }
        public string? Status { get; set; }
        public int? Num_episodes { get; set; }
        public virtual StartSeason? Start_season { get; set; }
        public virtual Broadcast? Broadcast { get; set; }
        public string? Source { get; set; }
        public int? Average_episode_duration { get; set; }
        public string? Rating { get; set; }
        public virtual ICollection<Studio>? Studios { get; set; }
    }
}

File: Frontend/src/app/interfaces/Anime.ts

export interface Anime {
    id: number;
    title?: string;
    main_picture?: MainPicture;
    alternative_titles?: AlternativeTitles;
    start_date?: string;
    end_date?: string;
    synopsis?: string;
    mean?: number;
    rank?: number;
    popularity?: number;
    num_list_users?: number;
    num_scoring_users?: number;
    nsfw?: string;
    genres?: Genre[] | null;
    created_at?: Date;
    updated_at?: Date;
    media_type?: string;
    status?: string;
    num_episodes?: number;
    start_season?: StartSeason;
    broadcast?: Broadcast;
    source?: string;
    average_episode_duration?: number;
    rating?: string;
    studios?: Studio[] | null;
}

export interface MainPicture {
    medium?: string;
    large?: string;
}

export interface AlternativeTitles {
    synonyms?: string[] | null;
    en?: string;
    ja?: string;
}

export interface Genre {
    id: number;
    name?: string;
}

export interface StartSeason {
    year?: number;
    season?: string;
}

export interface Broadcast {
    day_of_the_week?: string;
    start_time?: string;
}

export interface Studio {
    id: number;
    name?: string;
}

